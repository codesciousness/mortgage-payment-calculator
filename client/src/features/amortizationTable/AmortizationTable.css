.AmortizationTable {
    width: 600;
    height: 500;
}