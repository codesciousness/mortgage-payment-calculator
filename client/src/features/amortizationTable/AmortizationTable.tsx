import React from 'react';
import DataTable from '../../components/dataTable/DataTable';
import './AmortizationTable.css';

const columns = [
    {field: 'year', headerName: 'Year',  sortable: true, minWidth: 100, flex: 1},
    {field: 'principal', headerName: 'Principal', sortable: false, minWidth: 100, flex: 1},
    {field: 'interest', headerName: 'Interest', sortable: false, minWidth: 100, flex: 1},
    {field: 'remainingBalance', headerName: 'Remaining balance', type: 'number', sortable: false, minWidth: 150, flex: 1}
];

const AmortizationTable = (): JSX.Element => {

    return (
        <section id='AmortizationTable' className='AmortizationTable'>
            <h2>Amortization schedule breakdown</h2>
            <p>This table lists how much principal and interest are scheduled to be paid each year.</p>
            <DataTable rows={[]} columns={columns}/>
        </section>
    );
};

export default AmortizationTable;