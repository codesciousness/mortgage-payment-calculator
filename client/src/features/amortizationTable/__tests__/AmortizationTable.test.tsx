import React from 'react';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { store } from '../../../app/store';
import AmortizationTable from '../AmortizationTable';

describe('<AmortizationTable />', () => {
    test('AmortizationTable component is defined', () => {
        expect(<AmortizationTable/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(
            <Provider store={store}>
                <AmortizationTable/>
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly with initial state from Redux store', () => {
        render(
            <Provider store={store}>
                <AmortizationTable/>
            </Provider>
        );
        expect(screen.getByText('Amortization schedule breakdown')).toBeInTheDocument();
        expect(screen.getByText('Principal')).toBeInTheDocument();
        expect(screen.getByText('$328.88')).toBeInTheDocument();
        expect(screen.getByText('Interest')).toBeInTheDocument();
        expect(screen.getByText('$1,509.38')).toBeInTheDocument();
    });
    test('renders an export button with two export options when clicked', () => {
        render(
            <Provider store={store}>
                <AmortizationTable/>
            </Provider>
        );
        expect(screen.getByText('Export')).toBeInTheDocument();
        fireEvent.click(screen.getByText('Export'));
        expect(screen.getByText('Download as CSV')).toBeInTheDocument();
        expect(screen.getByText('Print')).toBeInTheDocument();
    });
    test('renders next page when Go to next page button is clicked', () => {
        render(
            <Provider store={store}>
                <AmortizationTable/>
            </Provider>
        );
        expect(screen.getByText('1–10 of 360')).toBeInTheDocument();
        expect(screen.queryByText('11–20 of 360')).not.toBeInTheDocument();
        fireEvent.click(screen.getByLabelText('Go to next page'));
        expect(screen.queryByText('1–10 of 360')).not.toBeInTheDocument();
        expect(screen.getByText('11–20 of 360')).toBeInTheDocument();
    });
});