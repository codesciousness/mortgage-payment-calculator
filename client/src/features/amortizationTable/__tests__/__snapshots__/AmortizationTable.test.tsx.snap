// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmortizationTable /> component matches the snapshot 1`] = `
<section
  className="AmortizationTable"
  id="AmortizationTable"
>
  <h2
    className="AmortizationTable__title"
  >
    Amortization schedule breakdown
  </h2>
  <p
    className="AmortizationTable__content"
  >
    This table lists how much principal and interest are scheduled to be paid each year.
  </p>
  <div
    aria-colcount={4}
    aria-multiselectable={false}
    aria-rowcount={360}
    className="MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard css-1lw6yti-MuiDataGrid-root"
    role="grid"
  >
    <div
      className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
    >
      <div
        className="MuiDataGrid-overlay css-18rq0oa-MuiDataGrid-overlay"
        error={[TypeError: Cannot set properties of null (setting 'scrollLeft')]}
      >
        An error occurred.
      </div>
    </div>
  </div>
</section>
`;
